tests/files/netcdf-file.cpp

```diff
@@ -9,37 +9,72 @@ using namespace chemfiles;
 
 
 TEST_CASE("Read a NetCDF file") {
-    NcFile file("data/netcdf/water.nc", File::READ);
-
-    CHECK(file.global_attribute("Conventions") == "AMBER");
-    // Usual dimmensions
-    CHECK(file.dimension("atom") == 297);
-    // Unlimited dimension
-    CHECK(file.dimension("frame") == 100);
-
-    CHECK(file.variable<nc::NcFloat>("cell_lengths").string_attribute("units") == "Angstrom");
-
-    auto var = file.variable<nc::NcFloat>("coordinates");
-    auto dims = var.dimmensions();
-    CHECK(dims.size() == 3);
-    CHECK(dims[0] == 100);
-    CHECK(dims[1] == 297);
-    CHECK(dims[2] == 3);
-
-    auto EPS = 1e-5f;
-    auto positions = var.get({0, 0, 0}, {1, 297, 3});
-    CHECK(std::abs(positions[0] - 0.4172191f) < EPS);
-    CHECK(std::abs(positions[1] - 8.303366f) < EPS);
-    CHECK(std::abs(positions[2] - 11.73717f) < EPS);
+    SECTION("Float variables") {
+        NcFile file("data/netcdf/water.nc", File::READ);
+
+        CHECK(file.global_attribute("Conventions") == "AMBER");
+        // Usual dimmensions
+        CHECK(file.dimension("atom") == 297);
+        // Unlimited dimension
+        CHECK(file.dimension("frame") == 100);
+
+        CHECK(file.variable<nc::NcFloat>("cell_lengths").string_attribute("units") == "Angstrom");
+
+        auto var = file.variable<nc::NcFloat>("coordinates");
+        auto dims = var.dimmensions();
+        CHECK(dims.size() == 3);
+        CHECK(dims[0] == 100);
+        CHECK(dims[1] == 297);
+        CHECK(dims[2] == 3);
+
+        auto EPS = 1e-5f;
+        auto positions = var.get({0, 0, 0}, {1, 297, 3});
+        CHECK(std::abs(positions[0] - 0.4172191f) < EPS);
+        CHECK(std::abs(positions[1] - 8.303366f) < EPS);
+        CHECK(std::abs(positions[2] - 11.73717f) < EPS);
+    }
+
+    SECTION("Double variables") {
+        NcFile file("data/netcdf/water.ncrst", File::READ);
+
+        CHECK(file.global_attribute("Conventions") == "AMBERRESTART");
+        // Usual dimmensions
+        CHECK(file.dimension("atom") == 297);
+
+        CHECK(file.variable<nc::NcDouble>("cell_lengths").string_attribute("units") == "angstrom");
+
+        auto var = file.variable<nc::NcDouble>("coordinates");
+        auto dims = var.dimmensions();
+        CHECK(dims.size() == 2);
+        CHECK(dims[0] == 297);
+        CHECK(dims[1] == 3);
+
+        auto EPS = 1e-5;
+        auto positions = var.get({0, 0}, {297, 3});
+        CHECK(std::abs(positions[0] - 0.4172191) < EPS);
+        CHECK(std::abs(positions[1] - 8.303366) < EPS);
+        CHECK(std::abs(positions[2] - 11.73717) < EPS);
+    }
 }
 
 TEST_CASE("Errors in NetCDF files") {
-    NcFile file("data/netcdf/water.nc", File::READ);
+    SECTION("Float variables") {
+        NcFile file("data/netcdf/water.nc", File::READ);
+
+        CHECK_THROWS_AS(file.global_attribute("FOO"), FileError);
```

Comment:
could you change these to `CHECK_THROW_WITH`? It helps making sure the exception thrown is the one intended

Reply:
Done.